sympy/core/tests/test_expr.py

```diff
@@ -2122,3 +2122,7 @@ def __repr__(self):
 
     assert (x == BadRepr()) is False
     assert (x != BadRepr()) is True
+
+def test_expr_free_symbols():
+    assert (2*exp(x)-3).expr_free_symbols == {x}
+    assert (4*y+3*x).expr_free_symbols == {x, y}
```

Comment:
```suggestion
    from sympy import IndexedBase
    assert pi.expr_free_symbols == {}
    assert x.expr_free_symbols == {x}
    assert (x + 2*y).expr_free_symbols == {x, y}
    i = Integral(x, (y, z))
    assert i.expr_free_symbols == i.function.free_symbols
    assert Function('f')(x).expr_free_symbols == {x}
    b = IndexedBase(x)[y]
    assert b.expr_free_symbols == {b}
```

Reply:
@smichr, the below examples are already `True` for master:
```py
    assert x.expr_free_symbols == {x}
    assert i.expr_free_symbols == i.function.free_symbols
    assert Function('f')(x).expr_free_symbols == {x}
    assert b.expr_free_symbols == {b}
```
Perhaps we can add like
```py
    assert Derivative(2*x, y).expr_free_symbols == {x}
    i = Integral(2*x, (y, z))
    assert i.expr_free_symbols == i.function.free_symbols
    assert Function('f')(x + 4).expr_free_symbols == {x}
    b = 4 * IndexedBase(x)[y]
    assert b.expr_free_symbols == {b.args[1]}
```
Kindly share your opinion.
Thanks.